Source/carthage/Printer.swift

```diff
@@ -0,0 +1,40 @@
+import Foundation
+
+public protocol Printer {
```

Comment:
I think that a unifying style means that `@testable import` shouldn't be used at all - you're not using it at all in `CarthageKitTests`, so why use it here?

I'm personally of the opinion that, unless you have good reason to, using `@testable import` means that you're lazy. You should write your implementation code as if it'll be consumed by some other entity (say... the tests). That is, `internal` should actually mean "private to the module", and `private` means "private to the file", instead of `internal` just meaning "I didn't want to type `public` in front of everything, but I also want to test my code".

In my own code/projects, I do use `@testable import` to test things like `ThreadSafePrinter`, which are implementation details that I just want to make sure I get right (and because it's the right thing to do - though `ThreadSafePrinter` in particular would be nigh-impossible to actually test correctly). `VersionCommand` just needs some instance of `Printer` to print to. When everything is integrated together in `carthage.app`, it just happens to use a `ThreadSafePrinter` (and, if I could, I would test that in main.swift, `VersionCommand` is init'd up with a `ThreadSafePrinter`, because that's important too).